src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -866,8 +881,11 @@ public Boolean call() throws InterruptedException {
             Boolean res;
             try {
                 res = results.get(0).get();
-            } catch (ExecutionException e) {
+            } catch (Throwable e) {
```

Comment:
I'm a little concerned about catching Throwable here. It's not generally a good idea. Is this one of those few places where Throwable makes sense? Would it be better to narrow it down to not catch catastrophic failures?

Reply:
yep, I will try to figure out which exceptions could be throw here, the main issue here is, if the exception is not catched, the retries start again from the max number of retries.